### ISSUE TITLE
Submodules support

### ISSUE THREAD

**Issue by jaxzin** [flynn-archive/gitreceived#12]
Are submodules supported? I've got [a project](https://github.com/jaxzin/api-samples-ruby-padrino/) that contains a git submodule reference to Twitter Bootstrap and I've deployed it successfully but the bootstrap.js is returning 404 from my service.

Possible issues:
- It's a old reference to 'twitter/bootstrap' instead of the more recent 'twbs/bootstrap' URL. So it may be choking on the HTTP redirect.
- I've got a symbolic link in my project ( [public/javascript/bootstrap](https://github.com/jaxzin/api-samples-ruby-padrino/blob/master/public/javascripts/bootstrap) ) which may be the actual failure and the submodule is a red herring.

**Comment by progrium** [comment]
Submodules are not supported. Dokku added submodule support and it was not
pretty. I don't want to add it here until I know a good way to do it.

On Wed, Jun 18, 2014 at 11:06 PM, Brian R. Jackson <notifications@github.com

> wrote:
> 
> Are submodules supported? I've got a project
> https://github.com/jaxzin/api-samples-ruby-padrino/ that contains a git
> submodule reference to Twitter Bootstrap and I've deployed it successfully
> but the bootstrap.js is returning 404 from my service.
> 
> Possible issues:
> - It's a old reference to 'twitter/bootstrap' instead of the more
>   recent 'twbs/bootstrap' URL. So it may be choking on the HTTP redirect.
> - I've got a symbolic link in my project ( public/javascript/bootstrap
>   https://github.com/jaxzin/api-samples-ruby-padrino/blob/master/public/javascripts/bootstrap
>   ) which may be the actual failure and the submodule is a red herring.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/flynn/gitreceived/issues/12.

## 

Jeff Lindsay
http://progrium.com

**Comment by titanous** [comment]
For reference, the Dokku implementation: progrium/dokku@7f0591fa1e99b4b92e56edc01b0651e4f955986c

Not pretty indeed.

**Comment by FlorianRappl** [comment]
I wanted to open a new issue, but I guess this would just be a duplicate: Submodules are not supported - also for buildpacks. I had to modify the [dotnet-core-buildpack](https://github.com/cloudfoundry/dotnet-core-buildpack) to include the compile-extensions directly; as the submodule was not auto-included.

What was tedious was not the manual inclusion, but rather the debugging process. Only after some manual investigation I found out what the real problem was.

Don't get me wrong - I love this project and I think you did a great job; also using buildpacks for the release process is great, at least as long as they work.

A simple enhancement would be to print a warning (or reference this in the documentation) when a `.gitmodules` file is found.

**Comment by titanous** [comment]
@FlorianRappl this repo is unmaintained, everything is now at https://github.com/flynn/flynn and I'm pretty sure submodules are fully supported for buildpacks and deployed repos.

**Comment by FlorianRappl** [comment]
@titanous Nope - did not work for me.

**Comment by titanous** [comment]
@FlorianRappl Okay, please open an issue on the flynn/flynn repo describing steps to reproduce the problem you ran into.
